packages/worker/src/spot-termination.ts

```diff
@@ -0,0 +1,52 @@
+import {
+	changeMessageVisibility,
+	logger,
+} from '@guardian/transcription-service-backend-common';
+import { SQSClient } from '@aws-sdk/client-sqs';
+import { getCurrentReceiptHandle, setInterruptionTime } from './index';
+
+const CHECK_FREQUENCY = 10;
+
+export const checkSpotInterrupt = async (
+	client: SQSClient,
+	queueUrl: string,
+) => {
+	const url = 'http://169.254.169.254/latest/meta-data/spot/instance-action';
+	try {
+		const result = await fetch(url);
+		if (result.status === 200) {
+			const json = await result.json();
+			if (json.action === 'terminate') {
+				const interruptionTime = new Date(json.time);
+				setInterruptionTime(interruptionTime);
+				logger.warn('Spot instance termination detected');
```

Comment:
shall we log termination time?
```suggestion
				logger.warn(`Spot instance interruption detected. Interruption time: ${interruptionTime}`);
```